### ISSUE TITLE
Test failure not accompanied by test name

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/TypeScript#18692]
**TypeScript Version:**  master

**Code**

Put `Debug.assert(node.fileName.indexOf("2dArrays") === -1);` as the first line of `checkSourceFile` in `checker.ts` and run `gulp runtests-parallel`.

**Expected behavior:**

Test failure includes the name of the failing test: `tests/cases/compiler/2dArrays.ts`.

**Actual behavior:**

```
Test worker encounted unexpected error and was forced to close:
        Message: Debug Failure. False expression.
        Stack: Error: Debug Failure. False expression.
    at checkSourceFile (/home/andy/TypeScript/src/compiler/checker.ts:22614:19)
    at Object.forEach (/home/andy/TypeScript/src/compiler/core.ts:186:32)
    at getDiagnosticsWorker (/home/andy/TypeScript/src/compiler/checker.ts:22718:13)
    at getDiagnostics (/home/andy/TypeScript/src/compiler/checker.ts:22681:24)
    at Object.getEmitResolver (/home/andy/TypeScript/src/compiler/checker.ts:528:13)
    at emitWorker (/home/andy/TypeScript/src/compiler/program.ts:1054:71)
    at /home/andy/TypeScript/src/compiler/program.ts:1007:51
    at runWithCancellationToken (/home/andy/TypeScript/src/compiler/program.ts:1130:24)
    at Object.emit (/home/andy/TypeScript/src/compiler/program.ts:1007:20)
    at Object.compileFiles (/home/andy/TypeScript/src/harness/harness.ts:1191:40)
    at /home/andy/TypeScript/src/harness/compilerRunner.ts:116:49
    at executeSuiteCallback (/home/andy/TypeScript/src/harness/parallel/worker.ts:57:23)
    at executeCallback (/home/andy/TypeScript/src/harness/parallel/worker.ts:71:13)
    at /home/andy/TypeScript/src/harness/parallel/worker.ts:59:56
    at Array.forEach (<anonymous>)
    at executeSuiteCallback (/home/andy/TypeScript/src/harness/parallel/worker.ts:59:18)
    at executeCallback (/home/andy/TypeScript/src/harness/parallel/worker.ts:71:13)
    at /home/andy/TypeScript/src/harness/parallel/worker.ts:17:56
    at Array.forEach (<anonymous>)
    at resetShimHarnessAndExecute (/home/andy/TypeScript/src/harness/parallel/worker.ts:17:18)
    at handleTest (/home/andy/TypeScript/src/harness/parallel/worker.ts:199:25)
    at process.<anonymous> (/home/andy/TypeScript/src/harness/parallel/worker.ts:161:41)
    at emitTwo (events.js:125:13)
    at process.emit (events.js:213:7)
    at process.emit (/home/andy/TypeScript/node_modules/source-map-support/source-map-support.js:439:21)
    at emit (internal/child_process.js:774:12)
    at _combinedTickCallback (internal/process/next_tick.js:141:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
```

May have been broken by #18462. CC @weswigham
